README.md

```diff
@@ -198,289 +189,494 @@ e.increment();  // logs 1
 e.increment();  // logs 2
 ```
 
-`@tracked` is defined in terms of the combination of two built-in decorators that we've seen before above. `@initialize` is used to replace the property definition with setting a property which stores the underlying data. `@register` is used to define a getter/setter pair which is used when accessing the property.
+When fields are decorated, the "wrapped" value is an object with two properties: `get` and `set` functions that manipulate the underlying storage. They are built to be `.call()`ed with the instance of the class as a receiver. The decorator can then return a new object of the same form. (If one of the callbacks is missing, then it is left in place unwrapped.)
 
 ```mjs
 // tracked.mjs
 
-export decorator @tracked {
-  @initialize(function(value, name) { this[`__internal_${name}`] = value; })
-  @register((target, name) => {
-    Object.defineProperty(target, "name", {
-      get() { return this[`__internal_${name}`]; },
-      set() { this[`__internal_${name}`] = value; this.render(); },
-      configurable: true
-    });
-  })
+export function tracked({get, set}) {
+  return {
+    get,
+    set(value) {
+      if (get.call(this) !== value) {
+        set.call(this, value);
+        this.render();
+      }
+    }
+  };
 }
 ```
 
-Note, further built-in decorators as in [NEXTBUILTINS.md](./NEXTBUILTINS.md#tracked) may provide a more direct and statically analyzable way to implement `@tracked` and avoid the use of `Object.defineProperty`.
-
-### `@bound`
-
-The `@bound` decorator makes a method auto-bound: it will carry around the original `this` value when accessed as `this.method` and not immediately called. This behavior matches Python's semantics, and it's been found useful in the React ecosystem, which makes frequent use of passing functions around. Example usage:
+This example could be roughly desugared as follows:
 
 ```mjs
-import { @bound } from "./bound.mjs";
+let initialize, get, set;
 
-class Foo {
-  x = 1;
+class Element {
+  #counter = initialize(0);
```

Comment:
Hmm, what would happen if I have two decorators with `initialize` function set? Will they be called in row? Like:

```js
class Element {
  @tracked2 @tracked1 counter = 0;  
}
```

```js
class Element {
  #counter = initialize2(initialize1(0))
}
```
Is this correct?

Reply:
Right, exactly, it would make a pipeline. (If someone wants to contribute an example of this, it'd be appreciated.)